engine/src/mbliphoneapp.mm

```diff
@@ -1717,63 +1717,140 @@ - (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection
-										   @"iPad 2(CDMA)",             @"iPad2,3",
-										   @"iPad 2(WiFi Rev A)",       @"iPad2,4",
-										   @"iPad Mini 1G (WiFi)",      @"iPad2,5",
-										   @"iPad Mini 1G (GSM)",       @"iPad2,6",
-										   @"iPad Mini 1G (GSM+CDMA)",  @"iPad2,7",
-                                           @"iPad 3(WiFi)",             @"iPad3,1",
-										   @"iPad 3(GSM+CDMA)",         @"iPad3,2",
-										   @"iPad 3(GSM)",              @"iPad3,3",
-										   @"iPad 4(WiFi)",             @"iPad3,4",
-                                           @"iPad 4(GSM)",              @"iPad3,5",
-										   @"iPad 4(GSM+CDMA)",         @"iPad3,6",
-                                        
-                                           @"iPad Air(WiFi)",        @"iPad4,1",
-                                           @"iPad Air(GSM)",         @"iPad4,2",
-                                           @"iPad Air(GSM+CDMA)",    @"iPad4,3",
-										   
-                                           @"iPad Mini 2G (WiFi)",       @"iPad4,4",
-                                           @"iPad Mini 2G (GSM)",        @"iPad4,5",
-                                           @"iPad Mini 2G (GSM+CDMA)",   @"iPad4,6",
-										   
-                                           @"iPod 1st Gen",         @"iPod1,1",
-                                           @"iPod 2nd Gen",         @"iPod2,1",
-                                           @"iPod 3rd Gen",         @"iPod3,1",
-										   @"iPod 4th Gen",         @"iPod4,1",
-                                           @"iPod 5th Gen",         @"iPod5,1",
-                                           // PM-2015-03-03: [[ Bug 14689 ]] Cast to NSString* to prevent EXC_BAD_ACCESS when in release mode and run in 64bit device/sim
-                                           (NSString *)nil];
-										   
+NSDictionary *commonNamesDictionary = [NSDictionary dictionaryWithObjectsAndKeys:
+									   @"iPhone",       @"iPhone1,1",
+									   @"iPhone 3G",    @"iPhone1,2",
+									   @"iPhone 3GS",   @"iPhone2,1",
+									   @"iPhone 4",     @"iPhone3,1",
+									   
+									   @"iPhone 4(Rev A)",      @"iPhone3,2",
+									   @"iPhone 4(CDMA)",       @"iPhone3,3",
+									   @"iPhone 4S",            @"iPhone4,1",
+									   @"iPhone 5(GSM)",        @"iPhone5,1",
+									   @"iPhone 5(GSM+CDMA)",   @"iPhone5,2",
+									   @"iPhone 5c(GSM)",       @"iPhone5,3",
+									   @"iPhone 5c(GSM+CDMA)",  @"iPhone5,4",
+									   @"iPhone 5s(GSM)",       @"iPhone6,1",
+									   @"iPhone 5s(GSM+CDMA)",  @"iPhone6,2",
+									   
+									   @"iPhone 6+ (GSM+CDMA)", @"iPhone7,1",
+									   @"iPhone 6 (GSM+CDMA)",  @"iPhone7,2",
+									   
+									   @"iPhone 6S",            @"iPhone8,1",
+									   @"iPhone 6S Plus",       @"iPhone8,2",
+									   @"iPhone SE",            @"iPhone8,4",
+									   @"iPhone 7 (CDMA)",      @"iPhone9,1",
```

Comment:
@runrevmark I am wondering, do we need to differentiate between e.g. `"iPhone 7 (CDMA)"`  and `"iPhone 7 (GSM)"`, or just a `iPhone 7` is enough for both cases? I guess the developer only cares if the device is an iPhone 7, not if it is GSM or CDMA.

Reply:
I'm not sure we should be mapping the model as reported by the system at all - i.e. just return things like iPhone11,2